haas/cli.py

```diff
@@ -265,6 +265,37 @@ def headnode_stop(headnode):
     url = object_url('headnode', headnode, 'stop')
     do_post(url)
 
+
+@cmd
+def node_register(node, subtype, *args):
+    """Register a node named <node>, with the given type
+        if obm is of type: ipmi then provide arguments
+        "ipmi", <hostname>, <ipmi-username>, <ipmi-password> 
+    """
+    obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+    obm_types = [ "ipmi", "mock" ]
+    #Currently the classes are hardcoded
+    #In principle this should come from api.py
+    #In future an api call to list which plugins are active will be added.
+
+
+    if subtype in obm_types:
+        if len(args) == 3:
+            obminfo = {"type": obm_api+subtype, "host": args[0],
+                        "user": args[1], "password": args[2]
+                      }
+        else:
+            sys.stderr.write('ERROR: subtype '+subtype+' requires exactly 3 arguments\n')
+            sys.stderr.write('<hostname> <ipmi-username> <ipmi-password>\n')
+            return
+    else:
+        sys.stderr.write('ERROR: Wrong OBM subtype supplied\n')
+        sys.stderr.write('Supported OBM sub-types: ipmi, mock\n')
+        return
+
+    url = object_url('node', node)
+    do_put(url, data={"obm": obminfo})
+
 @cmd
 def node_register(node, ipmi_host, ipmi_user, ipmi_pass):
```

Comment:
You need to actually remove the old version of this function. Did you test this? I would expect the second definition to mask the first, making this change not actually do anything.